src/bson.rs

```diff
@@ -3,3 +3,82 @@
 // Because BSON depends on the protocol version,
 // need to develop an Enum<Struct, Struct, ...> to
 // work based on the protocol version.
+use bson::spec::BinarySubtype;
+use bson::{bson, Binary, Bson, Document};
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+use serde::Deserialize;
+use std::io::{Error, ErrorKind};
```

Comment:
for the future i'd recommend separating std imports from other imports - but it isn't that big a deal and isn't a blocker here